src/js/color.ts

```diff
@@ -0,0 +1,3845 @@
+  }
+  const a = (c as number) * Math.cos(((h as number) * Math.PI) / (DEG * HALF))
+  const b = (c as number) * Math.sin(((h as number) * Math.PI) / (DEG * HALF))
+  const lms = transformMatrix(MATRIX_OKLAB_TO_LMS, [l as number, a, b])
+  const xyzLms = lms.map((cl) => Math.pow(cl, POW_CUBE))
+  const [x, y, z] = transformMatrix(MATRIX_LMS_TO_XYZ, xyzLms, true)
+  return [
+    'xyz-d65',
+    roundToPrecision(x!, HEX),
+    roundToPrecision(y!, HEX),
+    roundToPrecision(z!, HEX),
+    alpha,
+  ]
+}
+
+/**
+ * parse color()
+ * @param {string} value - color value
+ * @param {object} [opt] - options
+ * @param {string} [opt.colorSpace] - color space
+ * @param {boolean} [opt.d50] - xyz in d50 white point
+ * @param {string} [opt.format] - output format
+ * @returns {Array.<string|number>|?string}
+ *   - ['xyz-(d50|d65)', x, y, z, alpha], [cs, r, g, b, alpha], '(empty)', null
+ */
+export const parseColorFunc = (
+  value: string,
+  opt: {
+    colorSpace?: string
+    d50?: boolean
+    format?: string
+  } = {},
+): Array<string | number> | (string | null) => {
+  if (isString(value)) {
+    value = value.trim()
+  } else {
+    throw new TypeError(`${value} is not a string.`)
+  }
+  const { colorSpace, d50, format } = opt
+  if (!REG_FN_COLOR.test(value)) {
+    switch (format) {
+      case VAL_MIX: {
+        return null
+      }
+      case VAL_SPEC: {
+        return ''
+      }
+      default: {
+        return ['rgb', 0, 0, 0, 0]
+      }
+    }
+  }
+  const [, val] = value.match(REG_FN_COLOR) as [string, string]
+  // eslint-disable-next-line prefer-const
+  let [cs, v1, v2, v3, v4] = val.replace('/', ' ').split(/\s+/) as [
+    string,
+    string,
+    string,
+    string,
+    string,
+  ]
+  let r, g, b
+  if (cs === 'xyz') {
+    cs = 'xyz-d65'
+  }
+  if (v1 === NONE) {
+    r = 0
+  } else {
+    if (v1[0] === '.') {
+      v1 = `0${v1}`
+    }
+    r = v1.endsWith('%') ? parseFloat(v1) / MAX_PCT : parseFloat(v1)
+  }
+  if (v2 === NONE) {
+    g = 0
+  } else {
+    if (v2[0] === '.') {
+      v2 = `0${v2}`
+    }
+    g = v2.endsWith('%') ? parseFloat(v2) / MAX_PCT : parseFloat(v2)
+  }
+  if (v3 === NONE) {
+    b = 0
+  } else {
+    if (v3[0] === '.') {
+      v3 = `0${v3}`
+    }
+    b = v3.endsWith('%') ? parseFloat(v3) / MAX_PCT : parseFloat(v3)
+  }
+  const alpha = parseAlpha(v4)
+  if (REG_SPEC.test(format!) || (format === VAL_MIX && cs === colorSpace)) {
+    return [
+      cs,
+      v1 === NONE ? NONE : roundToPrecision(r, DEC),
+      v2 === NONE ? NONE : roundToPrecision(g, DEC),
+      v3 === NONE ? NONE : roundToPrecision(b, DEC),
+      v4 === NONE ? NONE : alpha,
+    ]
+  }
+  let x, y, z
+  // srgb
+  if (cs === 'srgb') {
+    ;[x, y, z] = convertRgbToXyz([r * MAX_RGB, g * MAX_RGB, b * MAX_RGB])
+    if (d50) {
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+      ;[x, y, z] = transformMatrix(MATRIX_D65_TO_D50, [x!, y!, z!], true)
+    }
+    // srgb-linear
```

Comment:
Is this also an automatic fix by code styling?
It would be more natural if it were not indented.

Reply:
Yeah, this is handled by Prettier. Mostly outside of our control